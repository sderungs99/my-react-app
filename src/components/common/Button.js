import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = (props) => {
    return (
        <button onClick={props.onClick}>{props.buttonLabel}</button>
    );
};

Button.propTypes = {
    buttonLabel: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
}

export default Button;