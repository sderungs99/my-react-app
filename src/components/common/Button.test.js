import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

const mockProps = {
    buttonLabel: "Test Button",
    onClick() {}
};

function setup(props) {
    return shallow(<Button {...props} />);
}

it('renders without crashing', () => {
    setup(mockProps);
});

it('renders as expected', () => {
    const component = renderer.create(
        <Button {...mockProps}/>
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

it('binds the onClick handler', () => {
    const spy = jest.spyOn(mockProps, 'onClick');
    let wrapper = setup(mockProps);
    wrapper.find('button').simulate('click');
    expect(spy).toHaveBeenCalled();
});