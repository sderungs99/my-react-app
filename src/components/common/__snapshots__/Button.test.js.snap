// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<button
  onClick={[Function]}
>
  Test Button
</button>
`;
